### PR TITLE
made a fix, inserted a test_api if required to test

### PR DESCRIPTION
**Pull request by Arghya115**

### CODE REVIEW

File: src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs [hunk 1]
```diff
@@ -66,16 +65,49 @@ public class BlockchainServiceWeb3 : JsonRpcService
...
+            Console.WriteLine("Tx count from block: "+  block.TransactionHashes.Count);
+            
+            ulong gasUsed = 0;
+            var txArray = new JArray{};
+            if(block.TransactionHashes.Count>0)
+            {
+                var txs = block!.TransactionHashes
+                .Select(hash => _transactionManager.GetByHash(hash)!)
+                .ToList();
+                Console.WriteLine("Entered here");
```
Comment (sgladkov) [inline src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs:100]: Don't use Console.WriteLine please,  change it to Logger calls if you need this records. But this is debug out put,. it is better to remove it at all our logs are to big already.

File: src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs [hunk 2]
```diff
@@ -57,7 +57,6 @@ public class BlockchainServiceWeb3 : JsonRpcService
             _systemContractReader = systemContractReader;
         }
 
-        [JsonRpcMethod("eth_getBlockByNumber")]
```
Comment (sgladkov) [inline src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs:60]: Why have you removed this attribute?

File: src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs [hunk 3]
```diff
@@ -66,16 +65,49 @@ public class BlockchainServiceWeb3 : JsonRpcService
...
-                : new JArray();
+                gasUsed = txs.Aggregate<TransactionReceipt, ulong>(0, (current, tx) => current + tx.GasUsed);
+
+                if(fullTx) txArray = Web3DataFormatUtils.Web3BlockTransactionArray(txs, block!.Hash, block!.Header.Index);
+            }
+
+            return Web3DataFormatUtils.Web3Block(block!, gasUsed, txArray);
+        }
+
+        [JsonRpcMethod("testapi")]
```
Comment (sgladkov) [inline src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs:84]: Don't add debug code to the release branch,  remove this function please

**Review by sgladkov — CHANGES_REQUESTED** [review]
Test it please before push

File: src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs [hunk 4]
```diff
@@ -66,13 +66,19 @@ public class BlockchainServiceWeb3 : JsonRpcService
             var block = _blockManager.GetByHeight((ulong)blockNumber);
             if (block == null)
                 return null;
-            var txs = block!.TransactionHashes
+
+            ulong gasUsed = 0;
+            var txArray = new JArray();
+            if(block.TransactionHashes.Count>0)
```
Comment (HMAshiqulIslam) [inline src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs:72]: Should we put an extra space before and after 0 ?

Comment (sgladkov) [reply src/Lachain.Core/RPC/HTTP/Web3/BlockchainServiceWeb3.cs:72]:
```suggestion
            if (block.TransactionHashes.Count > 0)
```
You can just accept the suggestion to fix

**Review by sgladkov — APPROVED** [review]
LGTM

**Pull request merged**